### PR TITLE
Implemented docs code block component 

### PR DESCRIPTION
**Pull request by aleksandrychev**

### CODE REVIEW

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/291/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/291/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-291/output/_site/

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/292/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/292/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-292/output/_site/

**Comment by aleksandrychev** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/294/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/294/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-294/output/_site/

File: cheatsheet.markdown [hunk 1]
```diff
@@ -208,6 +208,77 @@ however this does not support syntax highlighting and triple backticks are prefe
...
+
+These two blocks will be joined into one element on the UI.
+
+#### File code block
+
+You can specify file type of the code block by adding  `[file=Name of the file]` in the first line.
+This line won't be shown in the resulted HTML.
+
+```cf3
+[file=Agent.cf]
```
Comment (olehermanse) [inline cheatsheet.markdown:268]:
```suggestion
[file=policy.cf]
```

File: cheatsheet.markdown [hunk 2]
```diff
@@ -208,6 +208,77 @@ however this does not support syntax highlighting and triple backticks are prefe
...
+  ]
+}
+```
+
+These two blocks will be joined into one element on the UI.
+
+#### File code block
+
+You can specify file type of the code block by adding  `[file=Name of the file]` in the first line.
+This line won't be shown in the resulted HTML.
```
Comment (olehermanse) [inline cheatsheet.markdown:265]:
```suggestion
This line won't be shown in the resulting HTML (it will be converted to the heading / frame).
```

**Review by olehermanse — APPROVED** [review]
I left some smaller suggestions to the documentation above, looks good otherwise!

File: cheatsheet.markdown [hunk 3]
```diff
@@ -208,6 +208,77 @@ however this does not support syntax highlighting and triple backticks are prefe
...
+    }
+  ]
+}
+```
+
+These two blocks will be joined into one element on the UI.
+
+#### File code block
+
+You can specify file type of the code block by adding  `[file=Name of the file]` in the first line.
```
Comment (craigcomstock) [inline cheatsheet.markdown:264]:
```suggestion
You can specify file name of the code block by adding  `[file=Name of the file]` in the first line.
```

**Review by craigcomstock — COMMENTED** [review]
looks good other than a few suggestions

**Pull request merged**